### ISSUE TITLE
Add Pub/Sub Subscription#update

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/google-cloud-ruby#1283]
#1274 adds the API call to update subscriptions. Add this ability as the Subscription#update method.

**Comment by quartzmo** [comment]
@swcloud Should we label this issue `Status: Blocked`?

**Comment by swcloud** [comment]
ok

**Comment by blowmage** [comment]
Why is this blocked? What feature are we waiting on?

**Comment by blowmage** [comment]
@swcloud Is this issue still blocked? If not, can we start work on it?

@geigerj Do you have any input on this? All the other changes initiated by #1274 have already been implemented.

**Comment by geigerj** [comment]
@blowmage Please continue to hold off on this. There will be a batch of higher-priority changes to the pubsub client library surface coming in the next week or so that we should focus on first. (I'll have more details on those in the next few days.)

**Comment by blowmage** [comment]
@geigerj Is this something that should also be done now that we are actively working on Pub/Sub?

**Comment by geigerj** [comment]
Sure, we can move forward on this now, but if the other work is finished first and we are able to create a beta release, don't block on this feature. It has lower priority than the other Pub/Sub work because it is not a requirement for declaring the library beta.
